Zend/tests/match/043.phpt

```diff
@@ -0,0 +1,32 @@
+--TEST--
+Match expression error messages
+--FILE--
+<?php
+
+class Beep {}
+
+function test(mixed $var) {
+    try {
+        match($var) {};
+    } catch (UnhandledMatchError $e) {
+        print $e->getMessage() . PHP_EOL;
+    }
+}
+
+test(1);
+test(5.5);
+test("foo");
+test(true);
+test(false);
+test([1, 2, 3]);
+test(new Beep());
+
+?>
+--EXPECT--
+Unhandled match value: 1
+Unhandled match value: 5.5
+Unhandled match value: foo
```

Comment:
should be possibly quoted, otherwise empty strings will be very … empty. Similarly for nonprintables. Possibly have a look at `smart_str_append_escaped`.